File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.cpp
class Solution {
public:
  string reverseVowels(string s) {
    int i = 0;
    int j = s.size()-1;
    string vowels = "aeiouAEIOU";
    while (i < j) {
      i = s.find_first_of(vowels, i);
      j = s.find_last_of(vowels, j);
      if (i < j) {
        swap(s[i++], s[j--]);
      }
    }
    return s;
  }
};

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution:
  def reverseVowels(self, s: str) -> str:
    vowels = ('a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U')
    v = []
    for c in s:
      if c in vowels:
        v.append(c)
    ss = ""
    for i, c in enumerate(s):
      if c in vowels:
        ss += v[-1]
        v = v[:-1]
      else:
        ss += c
    return ss
      